helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/constraints/ConstraintBasedAlgorithm.java

```diff
@@ -57,11 +59,16 @@ class ConstraintBasedAlgorithm implements RebalanceAlgorithm {
   private static final Logger LOG = LoggerFactory.getLogger(ConstraintBasedAlgorithm.class);
   private final List<HardConstraint> _hardConstraints;
   private final Map<SoftConstraint, Float> _softConstraints;
+  private final Set<String> logEnabledClusters = new HashSet<>();
```

Comment:
This should be only one as per controller pipe line only serve for one cluster.

The constraint object is staying with per pipeline.

Reply:
Yes, got it. Updated the PR now, Junkai.